### ISSUE TITLE
False positive with E308 and mutable references

### ISSUE THREAD

**Issue by Heychsea** [intellij-rust/intellij-rust#5724]
## Environment

* **IntelliJ Rust plugin version:** 0.2.125.3191-201
* **Rust toolchain version:** 1.44.1
* **IDE name and version:** CLion 2020.1.2
* **Operating system:** macOS 10.15.5

## Problem description
Falsely reports E308 when cloning from a mutable reference to a variable expecting a non reference.

## Steps to reproduce
```rust
let mut a = vec![1];
let b = &mut a;
let c: Vec<Vec<_>> = vec![b.clone()]; // Generates a E308 error
```

**Comment by Undin** [comment]
Fixed in #5824
